### PR TITLE
✨ project listing with folder_id

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ✨ introduce `folder_id` query parameter in the project listing
- @odeimaiz tomorrow morning I will finish this PR https://github.com/ITISFoundation/osparc-simcore/pull/6132 just for Primary Group ID. And I think we might manage to connect it to frontend.  

## Related issue/s
- closes https://github.com/ITISFoundation/osparc-issues/issues/1567
- relates https://github.com/ITISFoundation/osparc-issues/issues/716

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__list_with_query_params.py [hunk 1]
```diff
@@ -360,3 +364,106 @@ async def test_list_projects_with_order_by_parameter(
...
+    with postgres_db.connect() as con:
+        output = []
+        result = con.execute(
+            folders.insert()
+            .values(
+                name="My Folder 1",
+                description="My Folder Decription",
+                product_name="osparc",
+                created_by=logged_user["primary_gid"],
+            )
```
Comment (sanderegg) [inline services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__list_with_query_params.py:384]: Please use the faker for the name and description.
for the product_name there is a fixture called product_name see [https://github.com/ITISFoundation/osparc-simcore/blob/ad25d00d8c12c91849932ba0c1920c616d36f182/packages/pytest-simcore/src/pytest_simcore/faker_products_data.py](https://github.com/ITISFoundation/osparc-simcore/blob/ad25d00d8c12c91849932ba0c1920c616d36f182/packages/pytest-simcore/src/pytest_simcore/faker_products_data.py)see if that can be re-used.

File: services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__list_with_query_params.py [hunk 2]
```diff
@@ -360,3 +364,106 @@ async def test_list_projects_with_order_by_parameter(
...
+        con.execute(
+            folders_to_projects.insert().values(
+                folder_id=_folder_id, project_uuid=user_project["uuid"]
+            )
+        )
+
+        yield FolderID(_folder_id)
+
+        con.execute(folders_to_projects.delete())
+        con.execute(folders.delete())
```
Comment (sanderegg) [inline services/web/server/tests/unit/with_dbs/02/test_projects_crud_handlers__list_with_query_params.py:398]: here you delete everything, not just what you created. is there any reason why? (e.g. keeping the folder_ids created and then deleting where the c.folder_id in folder_ids?) Also is there no cascading between folders and folders_to_projects?

**Review by sanderegg — APPROVED** [review]
thanks!

**Review by odeimaiz — APPROVED** [review]
thanks

**Review by GitHK — APPROVED** [review]

**Pull request merged**